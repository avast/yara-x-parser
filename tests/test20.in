//Global comment

//Rule comment
rule test
{
	//Rule block comment
	strings:
		$a = "string"

	//String comment
	condition:
		$a and 9 + 0 . "test"
}
