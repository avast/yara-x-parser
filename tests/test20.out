SOURCE_FILE@0..150
  COMMENT@0..16 "//Global comment"
  WHITESPACE@16..18 "\n\n"
  RULE@18..149
    COMMENT@18..32 "//Rule comment"
    WHITESPACE@32..33 "\n"
    RULE_KW@33..37 "rule"
    WHITESPACE@37..38 " "
    IDENTIFIER@38..42 "test"
    WHITESPACE@42..43 "\n"
    BLOCK_EXPR@43..149
      L_BRACE@43..44 "{"
      WHITESPACE@44..46 "\n\t"
      STRINGS@46..92
        COMMENT@46..66 "//Rule block comment"
        WHITESPACE@66..68 "\n\t"
        STRINGS_KW@68..75 "strings"
        COLON@75..76 ":"
        WHITESPACE@76..79 "\n\t\t"
        VARIABLE_STMT@79..92
          VARIABLE@79..81 "$a"
          WHITESPACE@81..82 " "
          ASSIGN@82..83 "="
          WHITESPACE@83..84 " "
          PATTERN@84..92
            STRING_LIT@84..92 "\"string\""
      WHITESPACE@92..95 "\n\n\t"
      CONDITION@95..147
        COMMENT@95..111 "//String comment"
        WHITESPACE@111..113 "\n\t"
        CONDITION_KW@113..122 "condition"
        COLON@122..123 ":"
        WHITESPACE@123..126 "\n\t\t"
        BOOLEAN_EXPR@126..147
          BOOLEAN_TERM@126..128
            VARIABLE@126..128 "$a"
          WHITESPACE@128..129 " "
          AND_KW@129..132 "and"
          WHITESPACE@132..133 " "
          BOOLEAN_EXPR@133..147
            BOOLEAN_TERM@133..147
              EXPRESSION@133..147
                PRIMARY_EXPR@133..134
                  INT_LIT@133..134 "9"
                WHITESPACE@134..135 " "
                PLUS@135..136 "+"
                WHITESPACE@136..137 " "
                EXPRESSION@137..147
                  PRIMARY_EXPR@137..138
                    INT_LIT@137..138 "0"
                  WHITESPACE@138..139 " "
                  DOT@139..140 "."
                  WHITESPACE@140..141 " "
                  PRIMARY_EXPR@141..147
                    STRING_LIT@141..147 "\"test\""
      WHITESPACE@147..148 "\n"
      R_BRACE@148..149 "}"
  WHITESPACE@149..150 "\n"
